core/deserialization.common/src/org/jetbrains/kotlin/serialization/deserialization/NameResolverUtil.kt

```diff
@@ -6,11 +6,18 @@
 package org.jetbrains.kotlin.serialization.deserialization
 
 import org.jetbrains.kotlin.metadata.deserialization.NameResolver
+import org.jetbrains.kotlin.metadata.deserialization.NameResolverImpl
 import org.jetbrains.kotlin.name.ClassId
 import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.name.FqName
 
 fun NameResolver.getClassId(index: Int): ClassId {
-    return ClassId.fromString(getQualifiedClassName(index), isLocalClassName(index))
+    if (this is NameResolverImpl) {
+        val identifiers = traverseIds(index)
```

Comment:
I'd prefer to add a method `fun getClassId(index: Int): String` to `NameResolver` instead, so that most of the other usages (which create a `ClassId` anyway) would also work more efficiently.

Since fixing most call sites of `getQualifiedClassName` would involve many changes, maybe it'd be better to extract such change into a separate PR.

Reply:
`ClassId` is in `kotlin.core.compiler.common` module. And `NameResolver` is in `kotlin.core.metadata`. The latter one does not depend on the first one. @udalov Do you think it might be more relevant to do it like described [here](https://github.com/JetBrains/kotlin/pull/5156#discussion_r1265343244) or to fix somehow module dependencies?